keps/sig-cloud-provider/2395-removing-in-tree-cloud-providers/README.md

```diff
@@ -195,153 +242,387 @@ import (
+The primary method of rollout failure is due to configuration issues. When
+migrating to use external CCMs there are several changes which must be made
+to ensure proper functioning. When encountering a failure, users should review
+the migration documentation to ensure that all command line flags, RBAC
+requirements, taint tolerations, and manifests are following the best
+practices.
+
+It is possible that a failed rollout could affect running workloads. Normal
+functioning of external CCMs involves activities around labeling new node
+objects, removing node objects, and configuring loadbalancer type services.
+If a rollout of external CCMs failed, users might see failures of workloads
+to schedule, workloads might be evicted, and service-based traffic could be
+interrupted.
+
+For more information about rollouts, upgrades, and planning, please see the
+Kubernetes documentation:
+* [Cloud Controller Manager Administration][ccmadmin]
+* [Migrate Replicated Control Plane To Use Cloud Controller Manager][ccmmigrate]
+
+###### What specific metrics should inform a rollback?
+There are a few observable behaviors that inform a rollback. If users start to
+see nodes failing to register, load balancer calls failing or not happening,
+deleted nodes not being removed, or components failing to start due to missing
+taint tolerations, these could all be possible signs that a CCM rollout has
+failed.
+
+In addition there are several metrics exposed by the cloud-provider framework
+that is maintained by SIG Cloud Provider. If the external CCM is using the
+cloud-provider framework, the following metrics could inform about failing
+behavior, users should be observant of these metric values increasing over
+a short period of time, or after upgrade:
+
+* `cloud_provider_taint_removal_delay_seconds`
+* `initial_node_sync_delay_seconds`
+* `nodesync_error_total`
+* `nodesync_latency_seconds`
+* `update_loadbalancer_host_latency_seconds`
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+Yes, the upgrade and downgrade paths have been tested manually on AWS and GCE
+by those provider owners. We do not have artifacts of those tests.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+* The provider options (e.g. `aws`, `azure`, etc), with the exception of `external`,
+  for the `--cloud-provider` flag of `kubelet`, `kube-apiserver`, and
+  `kube-controller-manager` are deprecated with the beta release.
+* The in-tree cloud controller loops for `kubelet`, `kube-apiserver`, and
+  `kube-controller-manager` are deprecated with the beta release.
+
+### Monitoring Requirements
 
-* **What are other known failure modes?**
+###### How can an operator determine if the feature is in use by workloads?
+* In genreal, if pods are being scheduled to the cluster then this feature is
```

Comment:
IIUC you wanted to capture the fact that credential provider works. If so, let's be more explicit that what we are looking for is successful fetches of container images.

Reply:
i've changed the wording on this one to make it more explicit.